pkgs/development/compilers/yosys/default.nix

```diff
@@ -43,23 +43,18 @@ stdenv.mkDerivation rec {
   };
 
   enableParallelBuilding = true;
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [ tcl readline libffi python3 bison flex protobuf zlib ];
 
   makeFlags = [ "ENABLE_PROTOBUF=1" "PREFIX=${placeholder "out"}"];
 
   patchPhase = ''
     patch -p1 < ${./plugin-search-dirs.patch}
     substituteInPlace ./Makefile \
-      --replace 'CXX = clang' "" \
-      --replace 'LD = clang++' 'LD = $(CXX)' \
-      --replace 'CXX = gcc' "" \
-      --replace 'LD = gcc' 'LD = $(CXX)' \
```

Comment:
As I recall it's necessary to use clang++ or g++ as LD.

Reply:
The makefile was re-written at some point to make all of these changes obsolete; the Makefile now 1) attempts `clang` by default and 2) if that fails, falls back to `CONFIG := gcc`. Later on, iff `$(CONFIG) == gcc`, then `LD = gcc` (just grep `CONFIG` in the Makefile.)

So this all works out, as it should.

Reply:
This sounds awful. LD ?= gcc would be better.

Reply:
You can specify whatever variables you want with a custom `Makefile.conf`, but again, in this case the build is fine (I think the actual reason is because the `LDFLAGS` in yosys specifies `stdc++` anyway so the implicit inclusion it gets with the `g++/clang++` commands is matched, even if you only use `gcc/clang` to compile/link.)

Reply:
This change broke compilation on darwin (see https://github.com/NixOS/nixpkgs/pull/112950/commits/5bf5284c21af27cdd870137cf76428f3e0b78a1b) or maybe it was already broken by an earlier change, I did not bisect that far. But if I re-add `--replace 'CXX = clang' "" \`, it compiles again. Why was this removed? Can I add it back? Should I only add this one line since that seems to be all that is needed? Should I make this a conditional that only adds this line on dariwn? I tested adding the line on NixOS and there were no changes in the output except for a different hash.